src/main/java/com/cse/helium/app/HeliumJavaApplication.java

```diff
@@ -12,7 +15,46 @@
 @EnableSwagger2WebFlux
 @ComponentScan("com.cse.helium")
 public class HeliumJavaApplication {
+
+  /**
+  * main.
+  */
   public static void main(String[] args) {
+    if (args != null) {
+      SimpleCommandLinePropertySource commandLinePropertySource =
+          new SimpleCommandLinePropertySource(args);
+      Arrays.stream(commandLinePropertySource.getPropertyNames()).forEach(s -> {
+        if (s.equals("log-level") || s.equals("l")) {
+          Configurator.setLevel("com.cse.helium",
+              setLogLevel(commandLinePropertySource.getProperty(s)));
+        }
+      });
+    }
+
     SpringApplication.run(HeliumJavaApplication.class, args);
   }
+
+  private static Level setLogLevel(String logLevel) {
+    switch (logLevel) {
+      case "trace":
+        return Level.TRACE;
+      case "debug":
+        return Level.DEBUG;
+      case "info":
+        return Level.INFO;
+      case "warn":
+        return Level.WARN;
+      case "error":
+        return Level.ERROR;
+      case "fatal":
+        return Level.FATAL;
+      default:
+        System.out.println("Usage: mvn clean spring-boot:run  "
+            + "-Dspring-boot.run.arguments=\"--h --auth-type=<CLI|MSI|VS>"
+            + " -keyvault-name=<keyVaultName>"
+            + "--log-level=<trace|info|warn|error|fatal>\"");
+        System.exit(0);
```

Comment:
shouldn't this be an exit(1) (or some other non-zero number) since it's a failure?
What is the default value? I believe we're using warn for the other languages

Reply:
Let's not make log level a fatal error.  We should get CLI args validation into one place and handle it there.  There is currently logic in EnvironmentReader for this.  Also, a setter shouldn't raise a fatal error, that needs to be done by the caller.  The caller should also decide on whether to use a default no argument or an invalid argument is passed.

Reply:
@jofultz If you specify the --log-level and specify an incorrect option, it should exit out. There is a default now (WARN) and also if you don't specify the option, it will continue with the default.

Also, there is a reason why the CLI args validation is not in one place. Springboot initializes the bean in a certain order. Keyvault etc is vailable at a different place in the initialization order than the "dry-run" requirement which is needed after the keyvault elements are initialized. Compared to that, setting the log level is needed right upfront as soon as the app starts so we don't miss logging from bean creation and app startup. It is unfortunate, but springboot operating model i.e. using automated bean creation and our logic and requirement i.e. specifying the keyvault name and env on cmdline vs "dry-run" to execute the keyvault values and then logging happen at different point  in lifetime of the app and thus don't lend to parsing all the params that we want upfront.

Reply:
I just checked commits and I don't see the default.  I see a System.exit(0) in the setLogLevel method.  The app should not exit in that method.

I also don't understand why validation can't be handled in the same place.